### PR TITLE
Spell check

### PR DESCRIPTION
**Pull request by ingydotnet**
Add `make check` to do various checks on the repo. 

Starting with a spell check based on `ispell`.
It uses a plain text personal dictionary which is committed as `spec/.spell`.

This depends on #133

### CODE REVIEW

**Comment by ingydotnet** [comment]
Need to dockerize ispell.

File: spec/.spell [hunk 1]
```diff
@@ -0,0 +1,214 @@
+american
+Antoniou
+API
+APIs
+app
+Aro
+as�
```
Comment (Thom1729) [inline spec/.spell:7]: There seems to be a problem with the encoding. The last byte in this line is `0xe2`.

Comment (ingydotnet) [reply spec/.spell]: I redid the whole thing to work with `aspell` instead. Works much nicer now.

File: spec/spec-changes.md [hunk 2]
```diff
@@ -12,15 +12,15 @@ specification version is **1.2.1**.
 * Clarify wording and examples
 * Removed sentences that are no longer relevant in 2021
 * Fixed known bugs in the grammar productions
-* More TODO
+* More TO DO
```
Comment (Thom1729) [inline spec/spec-changes.md:15]: This is a trivial point, but I think that “TODO” is more common and it's what I usually search for.

File: tool/bin/check-spec-file [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  spell-check "$file"
+)
+
+spell-check() (
+  printf "* Spell checking '%s' " "$file"
+
+  file=$1
+
+  out=$(aspell -p ./.aspell list < "$file")
```
Comment (Thom1729) [inline tool/bin/check-spec-file:16]: Would [`--mode markdown`](http://aspell.net/man-html/The-Options.html#Markdown-Filter) let us shrink the dictionary even more?

File: tool/docker/bin/check-spec-file [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+source "$YAML_SPEC_ROOT/tool/lib/runner.bash"
+
+check() (
+  command -v ispell >/dev/null || return 1
```
Comment (Thom1729) [inline tool/docker/bin/check-spec-file:6]: Should check for `aspell`?

Comment (ingydotnet) [reply tool/docker/bin/check-spec-file:6]: Good catch

File: tool/bin/check-spec-file (hunk 3 above)
Comment (ingydotnet) [reply tool/bin/check-spec-file:16]: yep!

File: spec/spec-changes.md (hunk 2 above)
Comment (ingydotnet) [reply spec/spec-changes.md:15]: :+1:

**Review by Thom1729 — APPROVED** [review]

**Pull request merged**